source/agora/consensus/protocol/Nominator.d

```diff
@@ -379,50 +387,18 @@ extern(D):
 
     /***************************************************************************
 
-        Restore SCP's internal state based on the provided ledger state
-
-        Params:
-            ledger = the ledger instance
+        Restore SCP's internal state based on the stored latest envelopes
 
     ***************************************************************************/
 
-    private void restoreSCPState (Ledger ledger)
+    protected void restoreSCPState ()
     {
-        import agora.common.Serializer;
-        import scpd.types.Stellar_SCP;
-        import scpd.types.Utils;
-        import scpd.types.Stellar_types : StellarHash = Hash, NodeID;
-        import std.range;
-
-        auto pub_key = NodeID(uint256(PublicKey(this.schnorr_pair.V[])));
-        auto block = ledger.getBlocksFrom(ledger.getBlockHeight()).front;
-        auto serialized = block.serializeFull();
-        auto vec = serialized.toVec();
-        auto value = duplicate_value(&vec);
-        SCPStatement statement =
+        foreach (const ref SCPEnvelope env; this.scp_envelope_store)
         {
-            nodeID: pub_key,
-            slotIndex: block.header.height,
-            pledges: {
-                type_: SCPStatementType.SCP_ST_EXTERNALIZE,
-                externalize_: {
-                    commit: {
-                        counter: 0,
-                        value: value,
-                    },
-                    nH: 0,
-                },
-            },
-        };
-
-        SCPEnvelope env = SCPEnvelope(statement);
```

Comment:
How about using the name of `envelope`. The `env' make me feel like 'environment'. But it depends on you.